File: auto_verbs/sites-not-suitable-SPA/shudder_com/shudder_search.js
const { Stagehand } = require("@browserbasehq/stagehand");
const { PlaywrightRecorder, setupLLMClient } = require("../../stagehand-utils");
const { z } = require("zod/v3");
const fs = require("fs");
const path = require("path");

const CFG = { genre: "supernatural", maxResults: 5 };

(async () => {
  const llmClient = setupLLMClient();
  const stagehand = new Stagehand({ env: "LOCAL", llmClient, headless: false });
  await stagehand.init();
  const recorder = new PlaywrightRecorder();
  const page = stagehand.context.pages()[0];
  try {
    const url = `https://www.shudder.com/`;
    await page.goto(url, { waitUntil: "domcontentloaded", timeout: 30000 });
    recorder.goto(url);
    await page.waitForTimeout(5000);

    await stagehand.act(`Search or browse for "${CFG.genre}" films or navigate to the supernatural category`);
    await page.waitForTimeout(5000);

    const { movies } = await stagehand.extract(
      `Extract the first ${CFG.maxResults} movies. For each get: title, year, director, duration, and description.`,
      z.object({
        movies: z.array(z.object({
          title: z.string().describe("Movie title"),
          year: z.string().describe("Release year"),
          director: z.string().describe("Director"),
          duration: z.string().describe("Duration"),
          description: z.string().describe("Movie description"),
        })),
      })
    );
    const items = movies.slice(0, CFG.maxResults);
    recorder.record("extract", { results: items });
    console.log("Extracted:", JSON.stringify(items, null, 2));
    fs.writeFileSync(path.join(__dirname, "recorded_actions.json"), JSON.stringify(recorder.actions, null, 2));
  } finally { await stagehand.close(); }
})();

File: auto_verbs/sites-not-suitable-SPA/shudder_com/shudder_search.py
"""Playwright script (Python) — Shudder"""
import os, sys, shutil
from dataclasses import dataclass, field
from typing import List
from playwright.sync_api import sync_playwright, Page
sys.path.insert(0, os.path.join(os.path.dirname(__file__), ".."))
from cdp_utils import get_free_port, get_temp_profile_dir, launch_chrome, wait_for_cdp_ws
from playwright_debugger import checkpoint

@dataclass(frozen=True)
class ShudderRequest:
    genre: str = "supernatural"
    max_results: int = 5

@dataclass
class MovieItem:
    title: str = ""
    year: str = ""
    director: str = ""
    duration: str = ""
    description: str = ""

@dataclass
class ShudderResult:
    movies: List[MovieItem] = field(default_factory=list)

def search_shudder(page: Page, request: ShudderRequest) -> ShudderResult:
    url = "https://www.shudder.com/"
    checkpoint("Navigate to Shudder")
    page.goto(url, wait_until="domcontentloaded", timeout=30000)
    page.wait_for_timeout(5000)
    result = ShudderResult()
    js_code = """(max) => {
        const results = [];
        const cards = document.querySelectorAll('[class*="movie"], [class*="title-card"], [class*="content-item"], article');
        for (const card of cards) {
            if (results.length >= max) break;
            const titleEl = card.querySelector('h2, h3, [class*="title"]');
            const title = titleEl ? titleEl.textContent.trim() : '';
            if (!title) continue;
            const descEl = card.querySelector('p, [class*="description"], [class*="synopsis"]');
            const description = descEl ? descEl.textContent.trim() : '';
            results.push({ title, year: '', director: '', duration: '', description });
        }
        return results;
    }"""
    for d in page.evaluate(js_code, request.max_results):
        item = MovieItem()
        item.title = d.get("title", "")
        item.description = d.get("description", "")
        result.movies.append(item)
    print(f"Found {len(result.movies)} movies")
    return result

def test_func():
    port = get_free_port()
    profile_dir = get_temp_profile_dir("shudder")
    chrome_proc = launch_chrome(profile_dir, port)
    ws_url = wait_for_cdp_ws(port)
    with sync_playwright() as pw:
        browser = pw.chromium.connect_over_cdp(ws_url)
        ctx = browser.contexts[0]
        page = ctx.pages[0] if ctx.pages else ctx.new_page()
        try:
            search_shudder(page, ShudderRequest())
        finally:
            browser.close()
            chrome_proc.terminate()
            shutil.rmtree(profile_dir, ignore_errors=True)

if __name__ == "__main__":
    from playwright_debugger import run_with_debugger
    run_with_debugger(test_func)
